api/src/main/java/org/openmrs/module/reporting/dataset/definition/RepeatPerTimePeriodDataSetDefinition.java

```diff
@@ -68,4 +79,23 @@ public void setRepeatPerTimePeriod(TimePeriod repeatPerTimePeriod) {
         this.repeatPerTimePeriod = repeatPerTimePeriod;
     }
 
+    /**
+     * @return the endDate
+     */
+    public DataSetDefinition getendDate() {
+        return endDate;
+    }
+
+    public void setEndDate(DataSetDefinition endDate){
+        this.endDate = endDate;
```

Comment:
check this error   Here

`incompatible types: org.openmrs.reporting.data.DatasetDefinition cannot be converted to org.openmrs.module.reporting.dataset.definition.DataSetDefinition.`

You have two different classes but same names .
Youll need to explicitly define which class you need  with a fully qualified path